File: code/HuaWeiNote/app/src/main/java/com/fan/hwnote/app/model/entity/Note.kt
package com.fan.hwnote.app.model.entity

/**
 * 笔记顶层实体。id == 0L 表示尚未持久化（NoteRepository.save 时插入）。
 *
 * Note 是不可变快照（用 .copy 改字段）；Block 内字段是 var（编辑器 View 直接改）。
 *
 * M12：新增 notebookId（null = "未分类"），与旧 categoryId 并存（Task 5 改造完毕后会下线 categoryId）。
 */
data class Note(
    val id: Long = 0L,
    val title: String = "",
    val plainText: String = "",
    val isFavorite: Boolean = false,
    val createdAt: Long = 0L,
    val updatedAt: Long = 0L,
    val content: NoteContent = NoteContent.empty(),
    val categoryId: Long? = null,     // null = 未分类
    val deletedAt: Long = 0L,         // 0 = 未删除；>0 = 删除时间戳
    val notebookId: Long? = null,     // M12：null = "未分类"
) {
    companion object {
        /** 新建一条笔记的内存对象。createdAt/updatedAt 都填 now，等待 save 时入库。 */
        fun new(now: Long = System.currentTimeMillis()): Note = Note(
            id = 0L,
            createdAt = now,
            updatedAt = now,
        )
    }
}

File: code/HuaweiNoteFlutter/lib/models/note.dart
import 'note_content.dart';

class Note {
  final int id;
  final String title;
  final String plainText;
  final bool isFavorite;
  final int createdAt;
  final int updatedAt;
  final NoteContent content;
  final int? categoryId;
  final int deletedAt;
  final int? notebookId;
  final String background;
  final String? firstImagePath;
  final bool hasTodo;

  Note({
    required this.id,
    this.title = '',
    this.plainText = '',
    this.isFavorite = false,
    required this.createdAt,
    required this.updatedAt,
    NoteContent? content,
    this.categoryId,
    this.deletedAt = 0,
    this.notebookId,
    this.background = 'plain',
    this.firstImagePath,
    this.hasTodo = false,
  }) : content = content ?? NoteContent.empty();

  factory Note.newNote({int? now}) {
    final ts = now ?? DateTime.now().millisecondsSinceEpoch;
    return Note(id: 0, createdAt: ts, updatedAt: ts);
  }

  Note copyWith({
    int? id, String? title, String? plainText, bool? isFavorite,
    int? createdAt, int? updatedAt, NoteContent? content, int? categoryId,
    bool setCategoryIdNull = false, int? deletedAt, int? notebookId,
    bool setNotebookIdNull = false, String? background,
    String? firstImagePath, bool clearFirstImagePath = false,
    bool? hasTodo,
  }) => Note(
    id: id ?? this.id,
    title: title ?? this.title,
    plainText: plainText ?? this.plainText,
    isFavorite: isFavorite ?? this.isFavorite,
    createdAt: createdAt ?? this.createdAt,
    updatedAt: updatedAt ?? this.updatedAt,
    content: content ?? this.content,
    categoryId: setCategoryIdNull ? null : (categoryId ?? this.categoryId),
    deletedAt: deletedAt ?? this.deletedAt,
    notebookId: setNotebookIdNull ? null : (notebookId ?? this.notebookId),
    background: background ?? this.background,
    firstImagePath: clearFirstImagePath ? null : (firstImagePath ?? this.firstImagePath),
    hasTodo: hasTodo ?? this.hasTodo,
  );
}
